### PR TITLE
Add capability to regenerate observed objective function values

### PR DESCRIPTION
**Pull request by gidden**
This PR adds a utility to generate nightly test observed values + documentation.

## How to review

- read docs
- try local execution

## PR checklist

- [x] Continuous integration checks all ✅
  
~- [ ] Add or expand tests; coverage checks both ✅~
- [x] Add, expand, or update documentation.
~- [ ] Update release notes.~

### CODE REVIEW

**Comment by gidden** [comment]
@khaeru - should now be good to go to support the next release

File: message_ix/tests/data/README.md
Comment (khaeru) [inline message_ix/tests/data/README.md:1]: There's an existing section describing these tests in [doc/contributing.rst](https://github.com/iiasa/message_ix/blob/main/doc/contributing.rst). Please either:
1. Expand that section, or
2. Create a new page within doc/contrib and link to it from contributing.rst, in the same way as the others.

File: message_ix/tests/data/generate_nightly_test_values.py
Comment (khaeru) [inline message_ix/tests/data/generate_nightly_test_values.py:13]: Please move this code into the existing file nightly.py:
https://github.com/iiasa/message_ix/blob/afefe8577b3decc16e1446ca3fc41efa72a10e4e/message_ix/testing/nightly.py#L105-L190

You can create a new subcommand of the "message-ix nightly" CLI command, e.g.:

```py
@cli.command("generate-values")
@click.pass_context
def generate_values(context):
   …
```

**Review by khaeru — CHANGES_REQUESTED** [review]
Thanks for these improvements. Two requests to make sure they are discoverable from our docs, and to keep code collected and easily usable.

**Comment by gidden** [comment]
I won't have time to make these improvements before the end of the week. If this blocks the release, then please simply take the scenarios.yaml file from this PR. Thanks!

**Comment by khaeru** [comment]
> I won't have time to make these improvements before the end of the week.

Can you @gidden or @OFR-IIASA now find time to complete the PR? If not, let's please raise in the weekly meeting so someone can be assigned to do that. It would be nice to fix the failures in our nightly tests that have been occurring for about 3 months, in order to unblock other work.

**Comment by gidden** [comment]
Unfortunately, I have not had time to do the docs changes. I will close this and PR only the minimal changes necessary.

**Pull request closed without merging**
